### ISSUE TITLE
NZBGet will not start after installing DSM 6.1.3-15152 Update 3

### ISSUE THREAD

**Issue by mchrisley** [SynoCommunity/spksrc#2880]
_For new Package Requests, see the guidelines_  

### Setup
_Package Name:_ NZBGet
_Package Version:_ 16.4-22

_NAS Model:_ DS1515+
_NAS Architecture:_ avoton (x86_64)
_DSM version:_ DSM 6.1.3-15152 Update 3

### Expected behavior
NZBGet should start

### Actual behavior
Failed to run package service
Also, error launching from CLI

### Steps to reproduce
_1._ Install DSM 6.1.3-15152 Update 3 on DS1515+
_2._ Reboot NAS or stop and start NZBGet
_3._ Try to launch NZBGet from package or CLI

I tried uninstalling and reinstalling the package, and it will not start even from a clean install.
/lib/libstdc++.so.6 does exist as a symlink:

root@Synology:/usr/local/nzbget/var# ls -al /lib/libstdc*
lrwxrwxrwx 1 root root     19 Jul 15 17:49 /lib/libstdc++.so.6 -> libstdc++.so.6.0.20
-r-xr-xr-x 1 root root 901208 Jul 12 18:58 /lib/libstdc++.so.6.0.20

#### Package log
_Check Package Center or `/usr/local/{package}/var/`_
```
One line created in nzbget-2017-08-18.log, only when triggered manually from CLI (none from package):
Fri Aug 18 11:03:57 2017    INFO    nzbget 16.4 daemon-mode
```
#### Other logs
_E.g. `/var/log/messages` 

2017-08-18T11:06:48-04:00 Synology [50232.573272] init: pkgctl-nzbget pre-start process (12130) terminated with status 1
2017-08-18T11:06:48-04:00 Synology synoscgi_SYNO.Core.Package.Control_1_start[11909]: servicecfg_internal_lib.c:187 Failed to start job [pkgctl-nzbget][0xD900 privilege.cpp:60]
2017-08-18T11:06:49-04:00 Synology synoscgi_SYNO.Core.Package.Control_1_start[11909]: service_initjob_action.c:27 InitJobStart failed, [0xD900 privilege.cpp:60]
2017-08-18T11:06:49-04:00 Synology synoscgi_SYNO.Core.Package.Control_1_start[11909]: pkgstartstop.cpp:279 Failed to start pkgctl-nzbget (err=-1) [0xD900 privilege.cpp:60]

`/var/log/synopkg.log`_
```
2017/08/18 11:06:46     start nzbget: begin to start version 16.4-22
2017/08/18 11:06:47     start nzbget 16.4-22 Begin pre-load apparmor
2017/08/18 11:06:47     start nzbget 16.4-22 End pre-load apparmor ret=[0]
2017/08/18 11:06:47     start nzbget 16.4-22 Begin start-stop-status start
Starting NZBGet ...
su: Permission denied
2017/08/18 11:06:48     start nzbget 16.4-22 End start-stop-status start ret=[1]
2017/08/18 11:06:48     (system) trigger nzbget 16.4-22 Begin start-stop-status stop
NZBGet is not running
2017/08/18 11:06:48     (system) trigger nzbget 16.4-22 End start-stop-status stop ret=[0]
2017/08/18 11:06:48     (system) trigger nzbget 16.4-22 Begin unload apparmor
2017/08/18 11:06:48     (system) trigger nzbget 16.4-22 End unload apparmor ret=[0]
2017/08/18 11:06:49     stop nzbget: begin to stop version 16.4-22
2017/08/18 11:06:49     stop nzbget: stop version 16.4-22 successfully, result 0
2017/08/18 11:06:49     start nzbget: start version 16.4-22 failed, result 1

CLI response when launched manually:
root@Synology:/volume1/@appstore/nzbget/var# ../bin/nzbget -D -c nzbget.conf
../bin/nzbget: /lib/libstdc++.so.6: no version information available (required by ../bin/nzbget)
../bin/nzbget: /lib/libstdc++.so.6: no version information available (required by ../bin/nzbget)
../bin/nzbget: /lib/libstdc++.so.6: no version information available (required by ../bin/nzbget)

One line created in nzbget-2017-08-18.log, only when triggered manually from CLI (none from package):
Fri Aug 18 11:03:57 2017    INFO    nzbget 16.4 daemon-mode

**Comment by mchrisley** [comment]
Here's what I get in /var/log/messages after manually adding the nzbget user to /etc/shadow

2017/08/18 11:40:23     start nzbget: begin to start version 16.4-22
2017/08/18 11:40:23     start nzbget 16.4-22 Begin pre-load apparmor
2017/08/18 11:40:23     start nzbget 16.4-22 End pre-load apparmor ret=[0]
2017/08/18 11:40:23     start nzbget 16.4-22 Begin start-stop-status start
Starting NZBGet ...
/usr/local/nzbget/bin/nzbget: /lib/libstdc++.so.6: no version information available (required by /usr/local/nzbget/bin/nzbget)
/usr/local/nzbget/bin/nzbget: /lib/libstdc++.so.6: no version information available (required by /usr/local/nzbget/bin/nzbget)
/usr/local/nzbget/bin/nzbget: /lib/libstdc++.so.6: no version information available (required by /usr/local/nzbget/bin/nzbget)
2017/08/18 11:40:23     start nzbget 16.4-22 End start-stop-status start ret=[0]
2017/08/18 11:40:23     start nzbget: start version 16.4-22 successfully, result 0

**Comment by ymartin59** [comment]
Hello. Just expected, as you installed a DSM 5.2 package on DSM6.1... at the moment repository contains NO DSM 6.x compatible packages.
I am building NZBGet for DSM 6.1 so that you may test it...

**Comment by ymartin59** [comment]
May you please install the following package manually and check if it works better ?
[nzbget_avoton-6.1_16.4-22.spk.zip](https://github.com/SynoCommunity/spksrc/files/1236294/nzbget_avoton-6.1_16.4-22.spk.zip)

**Comment by mchrisley** [comment]
Yes, I was able to install that package and have the clean install run. Thanks for your help!

**Comment by BenjV** [comment]
@ymartin59
I took a look at that package but it is not correctly made.

You are using the are using the "synogroup --add" command to add a username to the group sc-download.
This command will erase all users from that group and just leave the new username there.
So if you already have another user in that group it will be gone after this installation. 
You first have to read the existing users with the "synogroup --get" command and append the new name to that string and then use the "synogroup -add" command with the whole string of users.

Also in de the priviledge file you name the user of this package sc-nzbget and you also create a legacy user with the same name, that is not a wise decision, because that are totally different things.

In DSM 6 the package is running under the "system intenal name" which is set in the privilege file and that's not the same user as the one you create with the busybox.
This will confuse the users immensely, because on DSM 6 setting user permissions for the sc-nzbget user will not work for the package.
I would be wise to test for DSM 5 before adding a user with busybox, on DSM 6 that user won't have any use.

In this post I wrote a script how it could be done.
https://github.com/SynoCommunity/spksrc/issues/2216

**Comment by ymartin59** [comment]
@BenjV OK got it. I thought most packages have been already adapted to DSM 6 on branch `dsm6`.
Not sure I may have time to update package - I am focusing on `spkrepo` online service at the moment.

**Comment by BenjV** [comment]
Maybe they are all adapted, but then the adaption is done wrong.

**Comment by ymartin59** [comment]
As you are up to that point, may you please help and review each `spk` requiring a service account in branch `dsm6` and create an issue for those that are doubtful or obviously wrong ?

**Comment by BenjV** [comment]
The point is that you are using the installer.sh file on **every** package in the DSM 6 branch and all those installer.sh files have the mentioned problem.

If you like I can create try to make an installer.sh for you that will work and that can be adapted for all packages.

**Comment by ymartin59** [comment]
I would be really thankful if you have time to do so.

**Comment by BenjV** [comment]
I'll see what I can do to help.

**Comment by ymartin59** [comment]
Related to #2811

**Comment by Spike79** [comment]
I'm totally new here and trying to install any nzbget package on Dsm6. It seems people have been unable to install nzbget on Dsm6 since mid 2016?

**Comment by ymartin59** [comment]
nzbget package available in synocommunity repository is for DSM 5.x (and previous) version... so we can say DSM 6.x users have waited 3 months in average before complaining here.

**Comment by aniceberg** [comment]
I've been running nzbget 16.4-22 on a DS716+ (synology_braswell_ds716+) without issue until the latest update (DSM 6.1.3-15152 Update 7) where the package starts but stops a few seconds later. 

Is there any plan to release DSM6 compatible packages?

**Comment by Spike79** [comment]
I have the exact same versions of nzbget and DSM running, it doesn't seem to be an issue here. Even though I support your final question, maybe there is a way to get it running.

I'm no expert though. Did you change the user to root in the startstoprestart settings, or whatever it was called?


________________________________

**Comment by aniceberg** [comment]
I have the same versions of nzbget and DSM running on a DS415+ with no issue after the update. I figured it was due to the difference with the DS716+ architecture. No other changes were made to users/permissions.

I've combed through a few system logs but couldn't find anything entries related to nzbget other than what mchrisley shared above.

**Comment by Safihre** [comment]
Test package for new-style NZBGet has been released: #3138.
We no longer compile it here on the repo, but fetch the latest official installer from NZBGet.
During package installation you can choose if you want the latest stable or testing release. After this you can use NZBGet's internal update system to do updates. It also includes updates to user-management.
We hope to soon publish this on the official repo.
